### ISSUE TITLE
Invalid StackPanel size when Margin is set

### ISSUE THREAD

**Issue by wieslawsoltes** [Perspex/Perspex#432]
![stackpanelmargin](https://cloud.githubusercontent.com/assets/2297442/13127102/f2fed3aa-d5cd-11e5-8557-924f20a75a04.PNG)

Perspex Xaml
```XAML
<Window x:Class="MatrixPanAndZoomDemo.Perspex.MainWindow"
        xmlns="https://github.com/perspex"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:local="clr-namespace:MatrixPanAndZoomDemo.Perspex;assembly=MatrixPanAndZoomDemo.Perspex"
        UseLayoutRounding="True"
        Title="MainWindow" Height="600" Width="600">
    <Grid Background="SlateBlue">
        <StackPanel Background="#1FFFFFFF" Margin="4" HorizontalAlignment="Center" VerticalAlignment="Top">
            <TextBlock Text="E - Extent" Foreground="WhiteSmoke" FontSize="11"/>
            <TextBlock Text="F - Fill" Foreground="WhiteSmoke" FontSize="11"/>
            <TextBlock Text="R - Reset" Foreground="WhiteSmoke" FontSize="11"/>
            <TextBlock Text="Mouse Wheel - Zoom to Point" Foreground="WhiteSmoke" FontSize="11"/>
            <TextBlock Text="Mouse Right Button Down - Pan" Foreground="WhiteSmoke" FontSize="11"/>
        </StackPanel>
    </Grid>
</Window>
```

WPF Xaml
```XAML
<Window x:Class="MatrixPanAndZoomDemo.Wpf.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:local="clr-namespace:MatrixPanAndZoomDemo.Wpf"
        WindowStartupLocation="CenterScreen"
        UseLayoutRounding="True" SnapsToDevicePixels="True" TextOptions.TextFormattingMode="Display"
        Title="MainWindow" Height="600" Width="600">
    <Grid Background="SlateBlue">
        <StackPanel Background="#1FFFFFFF" Margin="4" HorizontalAlignment="Center" VerticalAlignment="Top">
            <TextBlock Text="E - Extent" Foreground="WhiteSmoke" FontSize="11"/>
            <TextBlock Text="F - Fill" Foreground="WhiteSmoke" FontSize="11"/>
            <TextBlock Text="R - Reset" Foreground="WhiteSmoke" FontSize="11"/>
            <TextBlock Text="Mouse Wheel - Zoom to Point" Foreground="WhiteSmoke" FontSize="11"/>
            <TextBlock Text="Mouse Right Button Down - Pan" Foreground="WhiteSmoke" FontSize="11"/>
        </StackPanel>
    </Grid>
</Window>
```

**Comment by wieslawsoltes** [comment]
Tested latest fix (build 9999.0.1330-nightly) and there are still issues when margin is set for StackPanel.

![przechwytywanie](https://cloud.githubusercontent.com/assets/2297442/14061525/24ef9f62-f382-11e5-949e-3b65e73dd1d8.PNG)

**Comment by wieslawsoltes** [comment]
Its fixed in 9999.0.1331-nightly.

![2016-03-26_20-10-14](https://cloud.githubusercontent.com/assets/2297442/14061964/ddf6961c-f38e-11e5-9a6a-84b2639c6f2f.png)
